sw/device/silicon_creator/lib/drivers/address_translation.c

```diff
@@ -0,0 +1,79 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/address_translation.h"
+
+#include "sw/device/lib/base/abs_mmio.h"
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+#include "sw/device/silicon_creator/lib/drivers/lifecycle.h"
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "otp_ctrl_regs.h"
+#include "rv_core_ibex_regs.h"
+
+enum {
+  kBase = TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR,
+};
+
+struct ibex_address_translation_regs {
```

Comment:
I think I'd be tempted to do this slot -> addr mapping in a switch statement in the function rather than using structs. Switch statements are easier to harden later if we need to.

Reply:
Done